### PR TITLE
Add the JSON logging format

### PR DESCRIPTION
**Pull request by fcoelho**
Fixes #?

Changes proposed in this pull request:
- Add a JSON format for logging

About the tests, I've basically copied what was there for `logfmt`, if some more comprehensive testing is needed, please let me know and I'll improve them.

Checklist:
- [x] docs updated
- [x] tests updated
- [x] [Changes](https://github.com/haraka/Haraka/blob/master/Changes.md) updated

### CODE REVIEW

File: tests/logger.js [hunk 1]
```diff
@@ -53,6 +53,22 @@ exports.log = {
...
+    'log in json' (test) {
+        this.logger.deferred_logs = [];
+        test.expect(3);
+        this.logger.format = this.logger.formats.JSON;
+        test.equal(0, this.logger.deferred_logs.length);
+        test.ok(this.logger.log('WARN','test warning'));
+        test.equal(1, this.logger.deferred_logs.length);
+        test.done();
+    },
+    'log in json w/deffered' (test) {
```
Comment (msimerson) [inline tests/logger.js]:
```suggestion
    'log in json w/deferred' (test) {
```

Comment (fcoelho) [reply tests/logger.js]: Changed this and the other "deffered" on the logfmt test too

**Comment by zuozp8** [comment]
config/log.ini should be updated, to show new possible configuration value

**Comment by msimerson** [comment]
Please explain. Or better, prepare a PR that shows what you think should be added.

**Comment by zuozp8** [comment]
https://github.com/haraka/Haraka/blob/master/config/log.ini#L10

in this case there should be:
```
; format=default, logfmt, json
```

possible configuration values should be enumerated in .ini files

**Pull request merged**